hypershift-operator/metrics_test.go

```diff
@@ -137,3 +147,69 @@ func TestMetrics(t *testing.T) {
 		})
 	}
 }
+
+func TestGetOperatorImage(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name            string
+		hypershiftPod   *corev1.Pod
+		expectedImage   string
+		expectedImageID string
+	}{
+		{
+			name: "When the pod is running it should get details from it",
+			hypershiftPod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "hypershift",
+				},
+				Status: corev1.PodStatus{
+					Conditions: []corev1.PodCondition{
+						{
+							Type:               corev1.PodReady,
+							Status:             corev1.ConditionTrue,
+							LastTransitionTime: metav1.Time{Time: time.Time{}.Add(10 * time.Minute)},
+						},
+					},
+					ContainerStatuses: []corev1.ContainerStatus{
+						{
+							Name:    assets.HypershiftOperatorName,
+							Image:   "quay.io/hypershift/hypershift:latest",
+							ImageID: "quay.io/hypershift/hypershift@sha256:746dc6b979cd6d42d4e6f3214e20eb5a2e0a2b4d1c1345f6914e5492e69d9afe",
+							State: corev1.ContainerState{
+								Running: &corev1.ContainerStateRunning{
+									StartedAt: metav1.Time{Time: time.Time{}},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedImage:   "quay.io/hypershift/hypershift:latest",
+			expectedImageID: "quay.io/hypershift/hypershift@sha256:746dc6b979cd6d42d4e6f3214e20eb5a2e0a2b4d1c1345f6914e5492e69d9afe",
+		},
+		{
+			name:            "When the pod is not running it should return strings with 'not found' and an error",
+			hypershiftPod:   &corev1.Pod{},
+			expectedImage:   "not found",
+			expectedImageID: "not found",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.hypershiftPod.Name = fakePodName
```

Comment:
>tc.hypershiftPod.Name = fakePodName

You don't need to do this with this version of the code, you can just harcode any arbitrary name in the spec